### PR TITLE
HTCONDOR-2761: dagman utils message printing

### PR DESCRIPTION
**Pull request by ColeBollig**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_utils/dagman_utils.cpp [hunk 1]
```diff
@@ -768,14 +802,13 @@ DagmanUtils::RenameRescueDagsAfter(const std::string& primaryDagFile, bool multi
 void
 DagmanUtils::tolerant_unlink(const std::string &pathname)
 {
+	if ( ! fileExists(pathname)) { return; }
```
Comment (Todd-L-Miller) [inline src/condor_utils/dagman_utils.cpp:805]: This does not prevent ENOENT, sadly; some other random process on the same machine could call `unlink()` between this check and the next.  You could revert back to the old code, except with `errnor != ENOENT` no special-case warning code?

File: src/condor_utils/dagman_utils.cpp [hunk 2]
```diff
@@ -944,21 +975,20 @@ DagmanUtils::create_lock_file(const char *lockFileName, bool abortDuplicates) {
 	if (result == 0 && abortDuplicates) {
 		int status;
 		if (ProcAPI::confirmProcessId(*procId, status) != PROCAPI_SUCCESS) {
-			dprintf(D_ERROR, "Warning: ProcAPI::confirmProcessId() failed; %d\n",
-			        status);
+			print_error("Warning: Failed to confirm process ID (%d)\n", status);
 		} else if ( ! procId->isConfirmed()) {
-			dprintf(D_ERROR, "Warning: ProcessId not confirmed unique\n");
+			print_error("Warning: ProcessId not confirmed unique\n");
```
Comment (Todd-L-Miller) [inline src/condor_utils/dagman_utils.cpp:980]: Not sure it make a lot of sense to have an error prefixed by "Warning".  You could maybe rephrase it as something like "Ignoring error that ProcessID not confirmed unique", I guess?

**Review by Todd-L-Miller — APPROVED** [review]
Looks good, aside from the ENOENT race condition, which you may decide isn't worth avoiding.  Good to see this get sorted out.

File: src/condor_utils/dagman_utils.cpp (hunk 1 above)
Comment (ColeBollig) [reply src/condor_utils/dagman_utils.cpp:805]: I figured this doesn't fully prevent `ENOENT`. I would rather treat a race condition deletion as an error message than a warning message at this point. It is a tolerant unlink and we detected the file existed prior to attempting to unlink so we should inform an error (In my opinion)

File: src/condor_utils/dagman_utils.cpp (hunk 2 above)
Comment (ColeBollig) [reply src/condor_utils/dagman_utils.cpp]: Added the `ignoring error that` portion to message and made message print to non-error stream

File: src/condor_utils/dagman_utils.cpp (hunk 1 above)
Comment (ColeBollig) [reply src/condor_utils/dagman_utils.cpp:805]: Also, I added the check so for file not existing at the top to prevent getting lots of failed to remove file because file does not exist messages for `condor_submit_dag` with a fresh DAG file and directory.

**Review by Todd-L-Miller — APPROVED** [review]
(still) LGTM.
